enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,442 @@
+
+authors:
+  - "@gabemontero"
+  - "@bparees" (provided a lot of input in tracking Jira)
+
+reviewers:
+  - "@bparees"
+  - "@adambkaplan"
+  - "@derekwaynecarr"
+  - "@zvonkok"
+  - "@who else?"
+  
+approvers:
+  - "@bparees"
+  - "@adambkaplan"
+  - "@derekwaynecarr"
+  - "@mrunalp"
+  - "@who else?"
+  
+creation-date: 2020-02-18
+
+last-updated: yyyy-mm-dd
+
+<!-- status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced -->
+status: provisional
+
+see-also:
+  - "/enhancements/builds/volume-secrets.md"  
+  
+---
+
+# A More Automatic Experience For Use of Entitled RHEL Content In OpenShift V4 Builds
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+ > 1. CRI-O already has config options (see the [config file doc](https://github.com/cri-o/cri-o/blob/master/docs/crio.conf.5.md)) 
+>to specify host directories that should always be mounted into containers
+>(i.e. `/etc/share/containers/mounts.conf`), akin to what the Red Hat Docker Daemon used to provide with OpenShift V3.  
+>Guidance in the early 4.x time frame has been around the cluster admin using the MCO post install to effectively update the
+>filesystem of each node with the necessary subscription related files in a well known location that CRI-O would look
+>for.  There is concern though using this through MCO from a couple of perspectives.  One, the MCO api is not
+>natural for someone who wants to do something higher level like "enable entitlements for builds".  Second, using the
+>MCO for any node update requires a restart of the node, which is onerous for this scenario.  Also note, the RHCOS
+>team want to disable their current default setting of auto-mounting secrets from the host (i.e. stuff from 
+>`/etc/share/containers/mounts.conf`), since RHCOS does not fully install subscription manager (and never will), and they have 
+>problems when only partial subscription manager metadata is available.  [RHCOS has the subscription-manager-certs 
+>installed but are missing the subscription-manger package which carries the rhsm.conf.](https://bugzilla.redhat.com/show_bug.cgi?id=1783393)
+>They are entertaining an MCO solution to work around this problem, but disabling auto-mount, and then having a 
+>solution which adds all the subscription manager in one fell swoop, would be their preference.  In the end, providing
+>an alternative to an MCO based solution **would seem** to facilitate RHCOS's future intentions and it would seem we should
+>minimally coordinate with them to facilitate how they stage future changes  
+> 2. There has been discussion elsewhere around having a more generic notion of "global resources" that any pod can
```

Comment:
Another thing that is really "annoying" for customers is the long turnaround time using MCO to lay down the files. Just a recent test at a customer site showed that we needed ~45 minutes to have those changes applied to three "high performance" nodes.  "Simple" nodes can reboot really fast, but having a node with specialized hardware and firmware a reboot takes a long time.

Reply:
interesting detail @zvonkok thanks ... that said, I've already got the text `...requires a restart of the node, which is onerous for this scenario`

For me, onerous == 45 minutes || reboot of any fashion

Going to leave this as is